lib/models/watch-detector.js

```diff
@@ -19,6 +19,26 @@ let POSSIBLE_WATCHERS = [
 
 let debug = require('heimdalljs-logger')('ember-cli:watcher');
 
+function watchmanVersionSatisfiesSemver(currentVersion, expectedSemanticVersion) {
+  let currentSatisfiesExpected = false;
+  try {
+    currentSatisfiesExpected = semver.satisfies(currentVersion, expectedSemanticVersion);
+  } catch (e) {
+    const dateStr = currentVersion
+      .replace(/\./g, '/')
+      .split('/')
+      .slice(0, -1)
+      .join('/');
+    const isValidDate = !isNaN(new Date(dateStr));
+    // preserve semver.satisfies behavior by rethrowing
+    if (!isValidDate) { throw e; }
+    // if current version can be parsed as a date,
+    // it is after the switch away from semver
+    currentSatisfiesExpected = true;
```

Comment:
do we need to specify a max date (that we rev occasionally) to prevent against future breaking changes?  Or, add it if/when that becomes an issue?